### PR TITLE
Refactoring cosmosdb module

### PR DESCRIPTION
**Pull request by tei-k**
# Description

https://scalar-labs.atlassian.net/browse/DLT-9322

# Done

- Add `use_kubernetes` flag for allow access from k8s node.
- Upgrade azure provider
- Add `allowed_cidrs` for firewall.

### CODE REVIEW

File: modules/azure/cosmosdb/main.tf [hunk 1]
```diff
@@ -7,16 +7,18 @@ resource "azurerm_cosmosdb_account" "db" {
 
   enable_automatic_failover = false
 
-  is_virtual_network_filter_enabled = true
+  is_virtual_network_filter_enabled = var.is_virtual_network_filter_enabled
```
Comment (ymorimo) [inline modules/azure/cosmosdb/main.tf:10]: Just a question. Is it needed to make this variable? Is there any case when we need to change this to `false`?

Comment (tei-k) [reply modules/azure/cosmosdb/main.tf:10]: Good point! I added it because it might be useful for debugging from local or antoher networks.

Comment (ymorimo) [reply modules/azure/cosmosdb/main.tf:10]: Ok. Then, `is_virtual_network_filter_enabled` is the same name as the `azurerm_cosmosdb_account`'s parameter, but as a scalar-terraform module, the name should probably be like `enable_virtual_network_filter`?

Comment (tei-k) [reply modules/azure/cosmosdb/main.tf]: @ymorimo 

The name `enable_virtual_network_filter ` looks good. Thank you! Fixed in 6eb29f4

**Review by ymorimo — APPROVED** [review]
LGTM!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**